### PR TITLE
Feature/dseat 25 docker

### PR DESCRIPTION
**Pull request by Ottorium**
https://re-motion.atlassian.net/browse/DSEAT-25

### CODE REVIEW

**Comment by Ottorium** [comment]
do we need a .dockerignore file?

File: RubeesEat/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build-env
+WORKDIR /App
+
```
Comment (a-ctor) [inline RubeesEat/Dockerfile:3]: Please add `EXPOSE 8080` to document that the image will listen to port 8080 TCP by default.

File: RubeesEat/RubeesEat.csproj [hunk 2]
```diff
@@ -6,4 +6,11 @@
         <ImplicitUsings>enable</ImplicitUsings>
     </PropertyGroup>
 
+    <ItemGroup>
+      <Content Update="Dockerfile">
+        <ExcludeFromSingleFile>true</ExcludeFromSingleFile>
+        <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
+      </Content>
+    </ItemGroup>
+
```
Comment (a-ctor) [inline RubeesEat/RubeesEat.csproj:15]: I don't think we need this. The `Dockerfile` is not necessary for the project output and thus does not need to be marked as `Content`.

File: RubeesEat/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+RUN dotnet publish -c Release -o out
+ 
+# Build runtime image
+FROM mcr.microsoft.com/dotnet/aspnet:8.0
+WORKDIR /App
+COPY --from=build-env /App/out .
+ENTRYPOINT ["dotnet", "RubeesEat.dll"]
+
+# to build: docker build -t <name> .
+# to run: docker run -p <port(use port above 1024)>:8080 <name> .
```
Comment (a-ctor) [inline RubeesEat/Dockerfile:18]: Personal pref. but I would put that up to to make it the first thing you see when opening the file

**Review by a-ctor — CHANGES_REQUESTED** [review]

File: RubeesEat/Dockerfile (hunk 1 above)
Comment (FelixSchoberRubi) [reply RubeesEat/Dockerfile:3]: We tried adding a port with this comment. Then we had to set a port number in Docker Desktop to run it. But it didn't work

Comment (FelixSchoberRubi) [reply RubeesEat/Dockerfile:3]: Never mind, now it works.
 we didn't use 8080 before, I guess that was the issue

File: RubeesEat/RubeesEat.csproj [hunk 4]
```diff
@@ -6,4 +6,5 @@
         <ImplicitUsings>enable</ImplicitUsings>
     </PropertyGroup>
 
+
```
Comment (a-ctor) [inline RubeesEat/RubeesEat.csproj:9]: ?

**Review by a-ctor — APPROVED** [review]

**Comment by a-ctor** [comment]
> do we need a .dockerignore file?

No, we do the build in its own container and only put the build output in the image that we push. As such, the image should already only contain the application files and nothing else.

**Pull request merged**